solutions/usaco-760.mdx

```diff
@@ -36,110 +36,69 @@ for i in past_order:
 	print(i)
 ```
 </PySection>
+	
 <CPPSection>
+	
 ```cpp
-// Created by Jesse Choe - Bronze Template
-
 #include <bits/stdc++.h>
 using namespace std;
-
-// Type aliases
-
-using ll = long long;
-using str = string;
-using vi = vector<int>;
-using pi = pair<int,int>;
-using vpi = vector<pi>;
-using si = set<int>;
-
-// Vector Operations
-
-#define sz (int)(x).size()
-#define pb push_back
-#define all(x) begin(x), end(x)
-#define sor(x) sort(all(x))
-#define rev(x) reverse(all(x))
-#define del(x, i) erase(begin(x)+i)
-#define rem(x,i) erase(begin(x), begin(x)+i)
-
-// Pairs
-
-#define mp make_pair
-#define sec second
-#define fir first
-
-// Sets and Maps
-
-#define ins insert
-#define ez erase
-#define cnt count
-
-// Math
-
-#define MOD 1e9+7
-#define MAX_INT 1e18*9
-
-// Permutation
-
-#define possibilities(x) while(next_permutation(all(x))
-
-// Loops
-
-#define Trav(a,x) for (auto& a: x)
-#define For(i,a,b) for (int i = (a); i < (b); ++i)
-
-// Input/Output - s is basically the file name without the ".in" and ".out"
-
-void setIO(string s) {
-	ios_base::sync_with_stdio(0); cin.tie(0);
-	freopen((s+".in").c_str(),"r",stdin);
-	freopen((s+".out").c_str(),"w",stdout);
-}
-
-int main(){
-
-	setIO("shuffle");
-
-	int N; cin >> N;
-
-	vi a(N);
-
-	For(i, 0, N){
-		cin >> a[i];
-	}
-
-	vi id(N);
-
-	For(i, 0, N){
-		cin >> id[i];
+const int MAX_N = 100;
+
+// Instantiating swapOrder array as global for easy access for custom find function 
+int swapOrder[MAX_N];
+int cowID[MAX_N];
+
+/*
+ * Returns the index of the element that contains j from the swapOrder array.
+ * We need the index to know where the j-th cow came from while we work backwards.
+ */
+int find(int n, int x) {
+	for (int i = 0; i < n; i++) {
+		if (swapOrder[i] == x)
+			return i;
 	}
+	return 0;
+}
 
-	// Perform the backwards swap
-
-	vi correct(N);
-
-	For(i, 0, 3){
-		for(int j=0; j<N; j++){
-			int idx;
-			for(int k=0; k<N; k++){
-				if(j+1 == a[k]){
-					idx = k;
-					break;
-				}
-			}
-			correct[idx] = id[j];
-		}
-		for(int j=0; j<N; j++){
-			id[j] = correct[j];
+int main() {
+	freopen("shuffle.in", "r", stdin);
+	freopen("shuffle.out", "w", stdout);
+	int n;
+	cin >> n;
+	for (int i = 0; i < n; i++) 
+		cin >> swapOrder[i];
+	for (int i = 0; i < n; i++) 
+		cin >> cowID[i];
+
+	/* 
+	 * For loop iterates 3 times and simulates 
+	 * each swap backwards during every iteration
+	 */
+	for (int i = 0; i < 3; i++) {
+		/* 
+		 * Temporary array to assist with the swapping 
+		 * as not to change the values of the cowID array.
+		 */
+		int temp[n];
```

Comment:
yeah i see the cowID is changed, but not this array

Reply:
oop sorry fixed now